### DISCUSSION TITLE
[Question][Hw3] llvm.dbg.value & warning

### DISCUSSION THREAD

**Discussion by esk223** [prosyslab-classroom/cs348-information-security#538] [Q&A]
1.
I implemented inserting call instructions, and here is my result.
entry:
  call void @llvm.dbg.value(metadata i32 10, metadata !11, metadata !DIExpression()), !dbg !12
  call void @__coverage__(i32 3, i32 14), !dbg !13
  %add = add nsw i32 10, 10, !dbg !13
  call void @llvm.dbg.value(metadata i32 %add, metadata !14, metadata !DIExpression()), !dbg !12
  call void @__coverage__(i32 4, i32 13), !dbg !15
  %sub = sub nsw i32 10, 10, !dbg !15
  call void @llvm.dbg.value(metadata i32 %sub, metadata !16, metadata !DIExpression()), !dbg !12
  call void @__dbz_sanitizer__(i32 %sub, i32 5, i32 13), !dbg !17
  call void @__coverage__(i32 5, i32 13), !dbg !17
  %div = sdiv i32 %add, %sub, !dbg !17
  call void @llvm.dbg.value(metadata i32 %div, metadata !18, metadata !DIExpression()), !dbg !12
  call void @__coverage__(i32 6, i32 3), !dbg !19
  ret i32 0, !dbg !19

The instruction says that "make sure that your tool does not instrument debug instructions (i.e., function calls to llvm.dbg.XXX)" but my result includes them. How can I remove them?

2.
When I run 'make', there are some warnings. Is this fine?
For example, I used the code "Some { filename; funname; line; column }" but "filename" and "funname" are not used, so it says "Warning 27: unused variable funname." I tried "Some { _; _; line; column }" but this gives an error. How can I fix this warning?

**Comment by hieudz** [comment]
1. I think the statement means you dont need to add __coverage__ before llvm.dbg.xxx. So your code is working correctly(?)
2. I guess that warning is fine. But if you want to solve just match your variable with Some(y) and then use y.line, y.column as needed

**Comment by esk223** [comment]
Got it. Thanks a lot!
